src/main/java/frc/robot/commands/vision/Align.java

```diff
@@ -0,0 +1,95 @@
+package frc.robot.commands.vision;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+import org.photonvision.EstimatedRobotPose;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class Align extends CommandBase{
+  private Drivetrain m_drive;
+  private Vision m_vision;
+  private double m_setpoint;
+  private double m_angle;
+  // private PIDController m_pid = new PIDController(1, 0.01, 0.1);
+
+  /**
+   * Aligns the robot to a specific angle
+   * @param angle The angle to go to
+   * @param drive The drivetrain
+   */
+  public Align(double angle, Drivetrain drive, Vision vision){
+    addRequirements(drive);
+    m_setpoint=angle;
+    m_drive=drive;
+    m_vision=vision;
+  }
+
+  private double getAngle(){
+    ArrayList<EstimatedRobotPose> p = m_vision.getEstimatedPoses(m_drive.getPose());
+    Pose2d p2;
+    if(p.size()==0){
+      return m_angle;
+    }else if(p.size()==1){
+      p2=p.get(0).estimatedPose.toPose2d();
+    }else{
+      p2 = new Pose2d(p.get(0).estimatedPose.getX()/2+p.get(1).estimatedPose.getX()/2, p.get(0).estimatedPose.getY()/2+p.get(1).estimatedPose.getY()/2, new Rotation2d(p.get(0).estimatedPose.toPose2d().getRotation().getRadians()/2+p.get(1).estimatedPose.toPose2d().getRotation().getRadians()/2));
```

Comment:
getRadians will output a value between -Pi and Pi so the avrageing radians might have a few probolems. For example the first pose has just under Pi andthe second is jut over -Pi. The avrage is around 0 but the acutal value is aproxamitly Pi or Pi.